hw03-quiz-app-spring-boot/src/main/java/ru/otus/homework/config/AppConfig.java

```diff
@@ -0,0 +1,25 @@
+package ru.otus.homework.config;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.context.properties.bind.ConstructorBinding;
+import org.springframework.stereotype.Component;
+
+@Getter
+@Setter
+@Component
+//@EnableConfigurationProperties(AppConfig.class)
+@ConfigurationProperties(prefix = "application")
+public class AppConfig implements LocaleConfig {
+
+    private int answersToPass;
+    private String locale;
+
+//    @ConstructorBinding //TODO: почему не работает такой конструктор? не видит аргументы answersToPass, locale
```

Comment:
Не видит Spring или не видит Idea?

Reply:
А фиг его знает, кто из них - падает при запуске c ошибкой 
***************************
APPLICATION FAILED TO START
***************************

Description:

AppConfig is annotated with @ConstructorBinding but it is defined as a regular bean which caused dependency injection to fail.

Action:

Update your configuration so that AppConfig is defined via @ConfigurationPropertiesScan or @EnableConfigurationProperties.

При этом @EnableConfigurationProperties на данный класс стоит.

Reply:
разобралась за несколько дней раскопок